### PR TITLE
Draft

### PR DESCRIPTION
**Pull request by longhornbuckle**

### CODE REVIEW

File: include/experimental/linalg/tensor_memory.hpp
Comment (2LoS) [inline include/experimental/linalg/tensor_memory.hpp:1]: I think we should work more on the tensor container's memory layout before posting a file dedicated to tensor_memory. The introduction of a fixed_allocator should be enough for now.

File: include/experimental/linalg/tensor.hpp
Comment (2LoS) [inline include/experimental/linalg/tensor.hpp:1]: If we want to present a working source code for the tensor container's implementation, we should work more on its features and descriptions to provide. The project seems to have already taken its own path, but it is a direction that is not completely defined. If I can, I would like to present an alternative version of the tensor.hpp file, that considers the descriptions in the tensor.bs file and contains updates on noexcept-specifications and allocators.

Comment (longhornbuckle) [reply include/experimental/linalg/tensor.hpp:1]: I wouldn't read much into the implementation. I can put them all on the .gitignore for the time being if you want. It mostly just parses - I doubt it really compiles. I've just been using as a scratch pad to figure out ideas. I don't expect to really have something working until we get more feedback.

I really just created a pull request to merge in so you'd have something to edit from.

**Comment by longhornbuckle** [comment]
I haven't figured out to get GitHub to restrict edits of a branch. Please do not edit this branch as each of us overwriting it defeats the point of git version control.

Once we've merged in acceptable changes (which requires your approval), we can then both create separate branches from main again and create pull requests of those new branches (tagging the other as a reviewer). This makes for an easier review process and makes managing differences and conflicts easier.

I've copied your change below so it is not lost when I remove your commit:

    After the call, the capacity is greater than or equal to the mapping_type(e).requested_span_size() if reallocation happens, and equal to the previous capacity otherwise. Reallocation happens at this point if and only if the current capacity is less than the mapping_type(e).requested_span_size().
     If an exception is thrown other than by the move constructor of a Cpp17NonCopyInsertable type, there are no effects.
     It throws length_error if max_size() < mapping_type(e).required_span_size().
     After successfully calling reserve, any resize( const extents_type& s ) for which the resultant s.extent(r) <= e.extent(r) for all r in the range[0,rank()) shall incur no re-allocation.
     
I strongly recommend running the bikeshed tool to update the .html file and view your changes:
curl https://api.csswg.org/bikeshed/ -F file=@tensor.bs -F force=1 > tensor.html

I don't think your changes will render the way you want them to.

**Review by 2LoS — APPROVED** [review]

**Review by 2LoS — APPROVED** [review]

**Review by 2LoS — APPROVED** [review]

**Pull request merged**
